upup/pkg/fi/cloudup/loader.go

```diff
@@ -214,6 +217,41 @@ func (l *Loader) addAssetCopyTasks(assets []*assets.ContainerAsset, lifecycle *f
 
 		}
 	}
+
+	return nil
+}
+
+func (l *Loader) addAssetFileCopyTasks(assets []*assets.FileAsset, lifecycle *fi.Lifecycle) error {
+	for _, asset := range assets {
+		glog.V(10).Infof("processing asset: %q, %q", asset.File, asset.CanonicalLocation)
+		if asset.CanonicalLocation != "" && asset.File != asset.CanonicalLocation {
+			context := &fi.ModelBuilderContext{
+				Tasks: l.tasks,
+			}
+
+			glog.V(10).Infof("adding task: %q", asset.File)
+			copyFileTask := &assettasks.CopyFile{
+				Name:       fi.String(asset.CanonicalLocation),
+				TargetFile: fi.String(asset.File),
+				SourceFile: fi.String(asset.CanonicalLocation),
+				Lifecycle:  lifecycle,
+			}
+
+			if asset.SHAValue != "" {
+				copyFileTask.SourceSha = fi.String(asset.SHAValue)
+			} else if asset.CononicalSHALocation != "" && asset.SHA != "" {
+				copyFileTask.SourceShaLocation = fi.String(asset.CononicalSHALocation)
+				copyFileTask.TargetSha = fi.String(asset.SHA)
+			}
+			if err := context.EnsureTask(copyFileTask); err != nil {
```

Comment:
Do we actually have duplicate tasks here, i.e. does this fail with AddTask?

Reply:
I can test on my integration branch.

Reply:
Yes I am